turbot/resource_turbot_resource.go

```diff
@@ -100,75 +112,104 @@ func resourceTurbotResourceCreate(d *schema.ResourceData, meta interface{}) erro
 	// assign the id
 	d.SetId(turbotMetadata.Id)
 	// save the formatted data: this is to ensure the acceptance tests behave in a consistent way regardless of the ordering of the json data
-	d.Set("data", helpers.FormatJson(d.Get("data").(string)))
+	if data, ok := d.GetOk("data"); ok {
+		d.Set("data", helpers.FormatJson(data.(string)))
+	}
 	if metadata, ok := d.GetOk("metadata"); ok {
 		d.Set("metadata", helpers.FormatJson(metadata.(string)))
 	}
+	if fullData, ok := d.GetOk("full_data"); ok {
+		d.Set("full_data", helpers.FormatJson(fullData.(string)))
+	}
+	if fullMetadata, ok := d.GetOk("full_metadata"); ok {
+		d.Set("full_metadata", helpers.FormatJson(fullMetadata.(string)))
+	}
 	d.Set("type", typeUri)
 	return nil
 }
 
 func resourceTurbotResourceRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*apiClient.Client)
 	id := d.Id()
-
-	// build required properties from data.
-	// properties is a map of property name -> property path
-
-	var properties map[string]string = nil
-
-	if _, ok := d.GetOk("data"); ok {
-		var err error = nil
-		properties, err = helpers.PropertyMapFromJson(d.Get("data").(string))
-		if err != nil {
-			return fmt.Errorf("error retrieving properties from resource data: %s", err.Error())
-		}
-	}
-
-	resource, err := client.ReadResource(id, properties)
+	var err error
+	// read full resource
+	resource, err := client.ReadFullResource(id)
 	if err != nil {
 		if apiClient.NotFoundError(err) {
 			// resource was not found - clear id
 			d.SetId("")
 		}
 		return err
 	}
-
-	// rebuild data from the resource
-	data, err := helpers.MapToJsonString(resource.Data)
-	if err != nil {
-		return fmt.Errorf("error building resource data: %s", err.Error())
+	// -`data` include the properties that are specified in the resource config
```

Comment:
// if data is set, only include the properties that are specified in the resource config

Reply:
also, move comment to after if statement to match the full_data comment

Reply:
make comments below consistent